lib/docker.coffee

```diff
@@ -1,9 +1,37 @@
 Docker = require 'dockerode'
 Promise = require 'bluebird'
+path = require('path')
+fs = require('mz/fs')
+url = require('url')
 
-docker = new Docker(socketPath: '/var/run/docker.sock')
-docker = Promise.promisifyAll(docker)
-# Hack dockerode to promisify internal classes' prototypes
-Promise.promisifyAll(Docker({}).getImage().constructor.prototype)
+# Cache the daemon
+docker = null
 
-module.exports = docker
+getDockerConnectOpts = ->
+	# Detect circleCi build
+	if process.env.DOCKER_HOST? and process.env.DOCKER_TLS_VERIFY?
+		certs = ['ca.pem', 'cert.pem', 'key.pem'].map((file) -> path.join(process.env.DOCKER_CERT_PATH, file))
+
+		Promise.map(certs, (c) -> fs.readFile(c))
```

Comment:
Unfortunately not, it interprets the `index` parameter as options, and then throws because it's a number.